test/e2e/test_catalogue_item.py

```diff
@@ -2,2198 +2,1539 @@
+                                                 would be required for a `CatalogueItemSchema`. Does not need mandatory
+                                                 IDs (e.g. manufacturer_id) as they will be added here.
+        """
+        # Where there are properties add the property ID, unit ID and unit value
+        expected_catalogue_item_get_data = E2ETestHelpers.add_property_ids_to_properties(
+            expected_catalogue_item_get_data, self.property_name_id_dict
+        )
+        properties = []
+        for prop in expected_catalogue_item_get_data["properties"]:
+            properties.append({**prop, **self._unit_data_lookup_dict[prop["id"]]})
+        expected_catalogue_item_get_data = {**expected_catalogue_item_get_data, "properties": properties}
+
+        return {
+            **expected_catalogue_item_get_data,
+            "catalogue_category_id": self.catalogue_category_id,
+            "manufacturer_id": self.manufacturer_id,
+        }
+
+    def post_catalogue_category(self, catalogue_category_data: dict) -> Optional[str]:
+        """
+        Posts a catalogue category with the given data and returns the ID of the created catalogue category if
+        successful.
+
+        :param catalogue_category_data: Dictionary containing the basic catalogue category data as would be required
+                                        for a `CatalogueCategoryPostSchema` but with any `unit_id`'s replaced by the
+                                        `unit` value in its properties as the IDs will be added automatically.
+        :return: ID of the created catalogue category (or `None` if not successful).
+        """
+        self.catalogue_category_id = CatalogueCategoryCreateDSL.post_catalogue_category(self, catalogue_category_data)
+
+        # Assign the property name id dict for any properties
+        if self.catalogue_category_id:
+            self.property_name_id_dict = {}
+            catalogue_category_data = self._post_response_catalogue_category.json()
+            for prop in catalogue_category_data["properties"]:
+                self.property_name_id_dict[prop["name"]] = prop["id"]
+                self._unit_data_lookup_dict[prop["id"]] = {"unit_id": prop["unit_id"], "unit": prop["unit"]}
+
+        return self.catalogue_category_id
+
+    def post_manufacturer(self, manufacturer_post_data: dict) -> Optional[str]:
+        """
+        Posts a manufacturer with the given data, returns the ID of the created manufacturer if successful.
+
+        :param manufacturer_post_data: Dictionary containing the manufacturer data as would be required for a
+                                       `ManufacturerPostSchema`.
+        :return: ID of the created manufacturer (or `None` if not successful).
+        """
+        self.manufacturer_id = ManufacturerCreateDSL.post_manufacturer(self, manufacturer_post_data)
+        return self.manufacturer_id
```

Comment:
Yeah we could save the lines here, but then every time we do a post we would also have to assign the ID, I thought this way was slightly nicer given the ids are left to the function itself rather than being present in the test which I thought would be less obvious to the person writing the test. We may use a test client as described in #90 which again could do the same thing but that is to be decided later.